app/src/main/java/com/asfoundation/wallet/ui/transact/TransferNavigator.kt

```diff
@@ -0,0 +1,125 @@
+                              amount: BigDecimal): Completable {
+    return Completable.fromAction {
+      val transaction = TransactionBuilder(TokenInfo(null, "Ethereum", "ETH", 18))
+      transaction.amount(amount)
+      transaction.toAddress(toWalletAddress)
+      transaction.fromAddress(walletAddress)
+      openConfirmation(transaction)
+    }
+  }
+
+  fun openAppcCreditsConfirmationView(walletAddress: String,
+                                      amount: BigDecimal,
+                                      currency: TransferFragmentView.Currency): Completable {
+    return Completable.fromAction {
+      val currencyName = when (currency) {
+        TransferFragmentView.Currency.APPC_C -> fragment.getString(
+            R.string.p2p_send_currency_appc_c)
+        TransferFragmentView.Currency.APPC -> fragment.getString(R.string.p2p_send_currency_appc)
+        TransferFragmentView.Currency.ETH -> fragment.getString(R.string.p2p_send_currency_eth)
+      }
+      fragmentManager.beginTransaction()
+          .replace(R.id.fragment_container,
+              AppcoinsCreditsTransferSuccessFragment.newInstance(amount, currencyName,
+                  walletAddress))
+          .commit()
+    }
+  }
+
+  private fun openConfirmation(transactionBuilder: TransactionBuilder) {
+    val intent = Intent(fragment.context, ConfirmationActivity::class.java).apply {
+      putExtra(C.EXTRA_TRANSACTION_BUILDER, transactionBuilder)
+    }
+    fragment.startActivityForResult(intent, TRANSACTION_CONFIRMATION_REQUEST_CODE)
+  }
+
+  fun navigateBack() = activity.onBackPressed()
+
+  fun hideKeyboard() {
+    val inputMethodManager =
+        activity.getSystemService(Activity.INPUT_METHOD_SERVICE) as InputMethodManager
+    var view = activity.currentFocus
+    if (view == null) view = View(activity)
+    inputMethodManager.hideSoftInputFromWindow(view.windowToken, 0)
+  }
+
+  fun showWalletBlocked() {
+    fragment.startActivityForResult(WalletBlockedActivity.newIntent(activity),
+        IabActivity.BLOCKED_WARNING_REQUEST_CODE)
+  }
+
+  fun showQrCodeScreen() {
+    val intent = Intent(activity, BarcodeCaptureActivity::class.java)
+    fragment.startActivityForResult(intent, BARCODE_READER_REQUEST_CODE)
+  }
+
+  fun showLoading() {
+    lockOrientation()
+    fragmentManager.beginTransaction()
+        .add(android.R.id.content, LoadingFragment.newInstance(), LoadingFragment::class.java.name)
+        .commit()
+  }
+
+  fun hideLoading() {
+    val fragment = fragmentManager.findFragmentByTag(LoadingFragment::class.java.name)
+    if (fragment != null) {
+      fragmentManager.beginTransaction()
+          .remove(fragment)
+          .commit()
+    }
+    unlockOrientation()
+  }
+
+  private fun lockOrientation() {
+    activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LOCKED
+  }
+
+  private fun unlockOrientation() {
+    activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_UNSPECIFIED
+  }
```

Comment:
Again, do these functions belong to the navigator? 
We should pass this to a view.